mailet/api/src/main/java/org/apache/mailet/Mail.java

```diff
@@ -251,8 +284,20 @@ default boolean hasSender() {
      *      if there was no such attribute (or if the attribute existed
      *      and its value was null)
      * @since Mailet API v2.1
+     * @deprecated see {@link #removeAttribute(AttributeName)
      */
+    @Deprecated
     Serializable removeAttribute(String name);
+
+    /**
+     * Removes the attribute with the given attribute name from this Mail instance.
+     * 
+     * @return the removed attribute, or null
```

Comment:
Sorry I missed this in my previous reviews, but ain't we supposed to no more design APIs using null?